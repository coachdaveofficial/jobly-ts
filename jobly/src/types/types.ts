export type Company = {
    handle: string;
    name: string;
    num_employees: number | null;
    description: string;
    logo_url: string | null;
};

export type Job = {
    id: number;
    title: string;
    salary: number | null;
    equity: number | null;
    company_handle: string;
};