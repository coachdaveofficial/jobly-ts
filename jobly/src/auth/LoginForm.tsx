import React, { useState } from 'react'
import './Login.css'
import { LoginFuncProp } from '../types/types'
import { useNavigate } from 'react-router-dom'
// import AuthRedirect from '../common/AuthRedirect'



export default function LoginForm({ login }: LoginFuncProp) {

    const INITIAL_STATE = {
        username: "testuser",
        password: "password"
    }
    const [formData, setFormData] = useState(INITIAL_STATE)
    const navigate = useNavigate();

    const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = evt.target;
        setFormData((prevData) => ({
            ...prevData,
            [name]: value
        }));
    }

    const handleSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
        evt.preventDefault();
        let result = await login(formData);
        if (result.success) {
            navigate('/companies');
        } else {
            alert(result.error);
        }
    }

    return (
        <>
            <form className='my-2 col-md-8 offset-md-2' onSubmit={handleSubmit}>

                <div className="login-container">
                    <label className='form-label' htmlFor="username"><b>Username</b></label>
                    <input
                        className='form-control'
                        id='username'
                        type="text"
                        placeholder="Enter Username"
                        name="username"
                        value={formData.username}
                        onChange={handleChange}
                        required
                    />

                    <label className='form-label' htmlFor="password"><b>Password</b></label>
                    <input
                        className='form-control'
                        id='password'
                        type="password"
                        placeholder="Enter Password"
                        name="password"
                        value={formData.password}
                        onChange={handleChange}
                        required />

                    <button className='btn btn-primary' type="submit">Login</button>

                </div>

                <div className="login-container" style={{ backgroundColor: '#f1f1f1' }}>
                    <button type="button" className="cancelbtn btn btn-danger" onClick={() => navigate('/')}>Cancel</button>
                </div>
            </form>
        </>

    )
}