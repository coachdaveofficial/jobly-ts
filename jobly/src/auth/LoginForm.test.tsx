import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from "react-router";
import { LoginData } from "../types/types";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <LoginForm login={function (data: LoginData): Promise<{ success: boolean; error?: any; }> {
        throw new Error("Function not implemented.");
      } } />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
