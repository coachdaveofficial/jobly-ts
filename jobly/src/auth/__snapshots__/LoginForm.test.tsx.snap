// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="my-2 col-md-8 offset-md-2"
  >
    <div
      class="login-container"
    >
      <label
        class="form-label"
        for="username"
      >
        <b>
          Username
        </b>
      </label>
      <input
        class="form-control"
        id="username"
        name="username"
        placeholder="Enter Username"
        required=""
        type="text"
        value="testuser"
      />
      <label
        class="form-label"
        for="password"
      >
        <b>
          Password
        </b>
      </label>
      <input
        class="form-control"
        id="password"
        name="password"
        placeholder="Enter Password"
        required=""
        type="password"
        value="password"
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Login
      </button>
    </div>
    <div
      class="login-container"
      style="background-color: rgb(241, 241, 241);"
    >
      <button
        class="cancelbtn btn btn-danger"
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</DocumentFragment>
`;
